pkg/apis/apps/v1alpha1/types.go

```diff
@@ -0,0 +1,103 @@
+type PetSet struct {
+	unversioned.TypeMeta `json:",inline"`
+	v1.ObjectMeta        `json:"metadata,omitempty"`
+
+	// Spec defines the desired identities of pets in this set.
+	Spec PetSetSpec `json:"spec"`
+
+	// Status is the current status of Pets in this PetSet. This data
+	// may be out of date by some window of time.
+	Status PetSetStatus `json:"status,omitempty"`
+}
+
+// A PetSetSpec is the specification of a PetSet.
+type PetSetSpec struct {
+	// Replicas is the desired number of replicas of the gieven Template.
+	// These are replicas in the sense that they are instantiations of the
+	// same Template, but individual replicas also have a consistent identity.
+	// TODO: Consider a rename of this field.
+	Replicas int `json:"replicas"`
+
+	// Selector is a label query over pods that should match the replica count.
+	// If empty, defaulted to labes on the pod template.
+	// More info: http://releases.k8s.io/HEAD/docs/user-guide/labels.md#label-selectors
+	Selector *unversioned.LabelSelector `json:"selector,omitempty"`
+
+	// Template is the object that describes the pod that will be created if
+	// insufficient replicas are detected. Each pod stamped out by the PetSet
+	// will fulfill this Template, but have a unique identity from the rest
+	// of the PetSet.
+	Template v1.PodTemplateSpec `json:"template"`
+
+	// VolumeClaims is a list of claims that pets are allowed to reference.
+	// The PetSet controller is responsible for mapping network identities to
+	// claims in a way that maintains the identity of a pet. Every claim in
+	// this list must have at least one matching (by name) volumeMount in one
+	// container in the template. A claim in this list takes precedence over
+	// any volumes in the template, with the same name.
+	// TODO: Define the behavior if a claim already exists with the same name.
+	VolumeClaims []v1.PersistentVolumeClaim `json:"volumeClaims,omitempty"`
+
+	// ServiceName is the name of the service that governs this PetSet.
+	// This service must exist before the PetSet.
+	// Pets get DNS/hostnames that follow the pattern:
+	// pet-specific-string.serviceName.default.svc.cluster.local
+	// where "pet-specific-string" is managed by the PetSet controller.
+	ServiceName string `json:"serviceName"`
+}
+
+// PetStatus represents the current state of a single pet in the set.
+type PetStatus struct {
+	// Name is the name of this particular pet.
+	Name string `json:"name,omitempty"`
+
+	// Volumes are the volumes associated with this pet.
+	Volumes []v1.Volume `json:"volumes,omitempty"`
+
+	// Role is the role of this pet in the cluster.
+	// All pets start off as non-members of an existing PetSet cluster. They
+	// become members when they flip to "Ready". This field might be updated
+	// by a babysitter controller that has more clarity about roles.
+	Role string `json:"role,omitempty"`
+}
+
+// PetSetStatus represents the current state of a PetSet.
+type PetSetStatus struct {
+	Pets []PetStatus `json:"pets,omitempty"`
```

Comment:
Do we actually need this?  In the proposal we stated we don't need to do this, I'd prefer the sets to be deterministic based on name and label selection and brian weighed in as not wanting to identify the set.